docs/_snippets/recipes_flutter_navigation/navigation2/main.dart.md

```diff
@@ -0,0 +1,135 @@
+```dart
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'bloc/book_bloc.dart';
+
+void main() {
+  Bloc.observer = SimpleBlocObserver();
+
+  runApp(
+    BlocProvider(
+      create: (context) => BookBloc(),
+      child: BooksApp(),
+    ),
+  );
+}
+
+class SimpleBlocObserver extends BlocObserver {
+  @override
+  void onEvent(Bloc bloc, Object? event) {
+    super.onEvent(bloc, event);
+    print(event);
+  }
+
+  @override
+  void onTransition(Bloc bloc, Transition transition) {
+    super.onTransition(bloc, transition);
+    print(transition);
+  }
+
+  @override
+  void onError(BlocBase bloc, Object error, StackTrace stackTrace) {
+    print(error);
+    super.onError(bloc, error, stackTrace);
+  }
+}
+
+class Book {
+  final String title;
+  final String author;
+
+  Book(this.title, this.author);
+}
+
+class BooksApp extends StatelessWidget {
+  final List<Book> books = [
+    Book('Left Hand of Darkness', 'Ursula K. Le Guin'),
+    Book('Too Like the Lightning', 'Ada Palmer'),
+    Book('Kindred', 'Octavia E. Butler'),
+  ];
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Books App',
+      home: BlocBuilder<BookBloc, BookState>(
+        builder: (context, state) {
+          return Navigator(
```

Comment:
In my opinion, it would be nice to use `FlowBuilder` here in order to simplify things.